### DISCUSSION TITLE
Support join as queries

### DISCUSSION THREAD

**Discussion by gudaoxuri** [SeaQL/sea-orm#557] [Q&A]
![image](https://user-images.githubusercontent.com/2802567/155629482-7255f9b2-5093-4729-ba82-763b24a51bc9.png)

There is a very common data structure: add creator Id and updater Id to some tables.
When querying these tables, it needs to return the creator name and the updater name.

I looked at the discussions (  #120 #251 #450 ) in sea-orm, and it says to use linked, but it doesn't seem to be possible to do the above query without supporting join aliases (join_as).

At the moment I can only achieve it through sea-query. I don't know how sea-orm should be implemented.

sea-query example:

```rust
let creator_table = Alias::new("creator");
let updater_table = Alias::new("updater");

Query::select()
            .columns(vec![
                (some_record::Entity, some_record::Column::Id),
                (some_record::Entity, some_record::Column::Data),
            ])
            .expr_as(Expr::tbl(creator_table.clone(), account::Column::Name), Alias::new("creator_name"))
            .expr_as(Expr::tbl(updater_table.clone(), account::Column::Name), Alias::new("updater_name"))
            .from(some_record::Entity)
            .join_as(
                JoinType::LeftJoin,
                account::Entity,
                creator_table.clone(),
                Expr::tbl(creator_table, account::Column::Id).equals(some_record::Entity, some_record::Column::CreatorId),
            )
            .join_as(
                JoinType::LeftJoin,
                account::Entity,
                updater_table.clone(),
                Expr::tbl(updater_table, account::Column::Id).equals(some_record::Entity, some_record::Column::UpdaterId),
            )
            .and_where(Expr::tbl(some_record::Entity, some_record::Column::Id).eq(id.to_string()))
```

**Comment by billy1624** [comment]
Hey @gudaoxuri, the example you provided was the right way of doing it, at the moment. In the near future we might add a feature for you to consolidate values of foreign column into single struct.

Related issue & PR:
- #540
- #415
